From fbb1bb006b7138ebeb842f200b3329f6a3e1d4b4 Mon Sep 17 00:00:00 2001
From: Riccardo Magliocchetti <riccardo.magliocchetti@gmail.com>
Date: Tue, 9 Nov 2010 22:21:11 +0100
Subject: [PATCH 20/34] Make modiying the processes we have to wait for list easier

Better let people modify the config file instead of poking with
with the shell script directly.
---
 bootchartd      |    8 ++++----
 bootchartd.conf |    3 +++
 2 files changed, 7 insertions(+), 4 deletions(-)

diff --git a/bootchartd b/bootchartd
index 47e98c6..b553368 100755
--- a/bootchartd
+++ b/bootchartd
@@ -38,6 +38,9 @@ AUTO_RENDER="no"
 AUTO_RENDER_DIR="/var/log"
 AUTO_RENDER_FORMAT="png"
 
+# The processes we have to wait for
+EXIT_PROC="kdm_greet xterm konsole gnome-terminal metacity mutter compiz ldm icewm-session enlightenment"
+
 # Read configuration.
 CONF="/etc/bootchartd.conf"
 if [ -f $PWD/bootchartd.conf ]; then
@@ -76,16 +79,13 @@ start()
 # Wait for the boot process to end.
 wait_boot()
 {
-	# The processes we have to wait for
-	local exit_proc="kdm_greet xterm konsole gnome-terminal metacity mutter compiz ldm icewm-session enlightenment"
-
 	# Wait for /proc first - without it we have issues
 	while [ ! -e /proc/cmdline ]; do
 		$USLEEP 5000
 	done
 
 	while true; do
-		if [ -n "$exit_proc" -a -n "$( pidof $exit_proc )" ]; then
+		if [ -n "$EXIT_PROC" -a -n "$( pidof $EXIT_PROC )" ]; then
 			# give an unambiguous settle afterwards - so we get
 			# more post-login data for slow systems
 			$USLEEP 20000000
diff --git a/bootchartd.conf b/bootchartd.conf
index 525c690..a9e4581 100644
--- a/bootchartd.conf
+++ b/bootchartd.conf
@@ -20,3 +20,6 @@ AUTO_RENDER_FORMAT="png"
 # Output directory for auto-generated boot charts
 AUTO_RENDER_DIR="/var/log"
 
+# The processes we have to wait for
+EXIT_PROC="kdm_greet xterm konsole gnome-terminal metacity mutter compiz ldm icewm-session enlightenment"
+
-- 
1.7.3.1

